CONTRIBUTING.md

```diff
@@ -1,59 +1,86 @@
 
-## Pull Request Process
+## Creating an issue.
 
-1. Create a branch. For example, if you are going to work on issue number #44 (which is, say, a new feature for ‘forgot password’ management):
+- Check if the issue you are going to propose is not duplicate of another issue.
+- Open a new issue according to type i.e., if issue is a bug open a new issue by clicking on `Get Started` in the scope of `Bug Report`.
+- Give a precise and meaningful name of the issue.
+- Describe your issue as good as possible that may ease the process of issue-reviewing by core-team.
+
+## How to contribute
+
+1. Fork the project and clone it to your local machine. Follow the [setup](https://github.com/CircuitVerse/CircuitVerse/blob/master/SETUP.md) guideline.
+2. Always take a pull from the remote repository to your master branch to keep it at par with the main project(updated repository).
+        
+        git pull upstream master
+        
+3. Create a branch. For example, if you are going to work on issue number #44 (which is, say, a new feature for ‘forgot password’ management):
 
         git checkout -b forgot-password#44
 
     This both creates and checks out that branch in one command.  
     The feature name should provide a (short) description of the issue.
 
-2. Update the README.md with details of changes to the interface, this includes new environment variables, exposed ports, useful file locations and container parameters.
-   
-3. Develop the feature and push it to your local repo.
-   
-4. Create Pull Request (PR). Make sure to comment the issue that your PR is supposed to solve.
+4. Update the README.md with details of changes to the interface, this includes new environment variables, exposed ports, useful file locations and container parameters.
+5. Develop the feature and push it to your local repo.
+6. Create Pull Request (PR). Make sure to comment the issue that your PR is supposed to solve.
 
-## Community
-Discussions about CircuitVerse issues and features take place on the repository's [Issues](https://github.com/CircuitVerse/CircuitVerse/issues) and [Pull Requests](https://github.com/CircuitVerse/CircuitVerse/pulls) sections. Anybody is welcome to join these conversations. See the [README](README.md) for more information on communication channels.
+## Create a pull request
+
+- Try to keep the pull requests small. A pull request should try its very best to address only a single concern.
+- For work in progress pull requests, please use the Draft PR feature.
+- Make sure all tests pass and add additional tests for the code you submit.
+- Document your reasoning behind the changes. Explain why you wrote the code in the way you did. The code should explain what it does.
+- If there's an existing issue, reference to it by adding something like `References/Closes/Fixes/Resolves #123`, where 123 is the issue number. 
+- Please fill out the PR Template when making a PR.
```

Comment:
Also mention when to close a PR discussion as resolved, especially one generated by static code tools lilke codeclimate or hound

Reply:
Done